Modules/posixmodule.c

```diff
@@ -5586,17 +5585,20 @@ os__path_abspath_impl(PyObject *module, path_t *path)
 
         if ((size_t)abs_len + 1 > (size_t)PY_SSIZE_T_MAX / sizeof(wchar_t)) {
             PyErr_SetString(PyExc_OverflowError, "path is too long");
+            PyMem_Free(cwd_buf);
```

Comment:
Free memory before calling the `PyErr` APIs. Ditto below.

Reply:
Fixed.

Reply:
It has been moved, I believe you reopened this.